src/main/java/jenkins/scm/api/SCMSourceCriteria.java

```diff
@@ -51,7 +51,7 @@
     /**
      * A probe for a branch candidate. Inspectors can tell whether a file path exists.
      */
-    static abstract class Probe implements Serializable {
+    abstract class Probe implements Serializable {
```

Comment:
Did you really need to make this non`static`? Will break serial form for sure, and maybe binary compatibility too.

If you really need a reference to `SCMSourceCriteria` for some reason, it is better to introduce an explicit, nullable field with an injector method, so that you can consciously control compatibility issues.